### PR TITLE
Add validation for env var format

### PR DESCRIPTION
**Pull request by adam2k**
The environment variables can only be uppercase A-Z and underscores.

In reference to: https://github.com/astronomer/issues/issues/856

### CODE REVIEW

File: src/lib/deployments/validate/environment/index.unit.test.js [hunk 1]
```diff
@@ -8,10 +8,46 @@ describe("When valid environment variables are passed", () => {
...
+    expect(() => {
+      const badEnvs = [
+        { key: "ONE_VALID", value: ""},
+        { key: "TWO_VALID", value: ""},
+        { key: "LAST IS INVALID", value: "" }
+      ];
+      validateEnvironment(badEnvs);
+    }).toThrow();
+  });
+});
```
Comment (andriisoldatenko) [inline src/lib/deployments/validate/environment/index.unit.test.js:53]:
```suggestion
});

```

**Comment by andriisoldatenko** [comment]
@adam2k `npm run lint` fails

**Comment by andriisoldatenko** [comment]
Can you please update codecov report to 77%?

File: src/lib/deployments/validate/environment/index.js [hunk 2]
```diff
@@ -35,3 +40,16 @@ export default function validateEnvironment(envs) {
     throw new InvalidDeploymentError(msg);
   }
 }
+
+function validateEnvironmentVarialbles(envs) {
```
Comment (andriisoldatenko) [inline src/lib/deployments/validate/environment/index.js:44]: @adam2k can you please add comment to describe your function?
something like:
https://github.com/astronomer/houston-api/blob/d111c9f4dfb10e8eeac67a23412958a4d3e8b4b3/src/lib/deployments/validate/index.js#L9-L13

File: src/lib/deployments/validate/environment/index.unit.test.js [hunk 3]
```diff
@@ -8,10 +8,46 @@ describe("When valid environment variables are passed", () => {
...
       const badEnvs = [{ key: "AIRFLOW__CORE__EXECUTOR", value: "" }];
       validateEnvironment(badEnvs);
     }).toThrow();
   });
+
+  test("it throws an error for invalid key formats", () => {
+    expect(() => {
+      const badEnvs = [{ key: "", value: "" }];
+      validateEnvironment(badEnvs);
+    }).toThrow();
```
Comment (andriisoldatenko) [inline src/lib/deployments/validate/environment/index.unit.test.js:22]:
```
expect(() => {
      const badEnvs = [{ key: "_", value: "test" }];
      validateEnvironment(badEnvs);
    }).toThrow();
expect(() => {
      const badEnvs = [{ key: "__", value: "test" }];
      validateEnvironment(badEnvs);
    }).toThrow();

```

File: src/lib/deployments/validate/environment/index.js (hunk 2 above)
Comment (schnie) [reply src/lib/deployments/validate/environment/index.js:52]: `validateEnvironmentVarialbles` looks like a typo :)

File: src/lib/deployments/validate/environment/index.unit.test.js (hunk 3 above)
Comment (adam2k) [reply src/lib/deployments/validate/environment/index.unit.test.js:22]: Are you suggesting that the env var should not end in an underscore?  OR do you think that the case is we need at least one uppercase character so something like `_A_` is valid?

Comment (adam2k) [reply src/lib/deployments/validate/environment/index.unit.test.js:32]: I updated the regex and added tests.  I'm assuming starting and ending with _ is OK.

Comment (andriisoldatenko) [reply src/lib/deployments/validate/environment/index.unit.test.js:32]: nope i demonstrated that env var `_` like this is invalid.

File: src/lib/deployments/validate/environment/index.unit.test.js [hunk 4]
```diff
@@ -5,13 +5,81 @@ describe("When valid environment variables are passed", () => {
...
+
+    expect(() => {
+      const badEnvs = [{ key: "_", value: "test" }];
```
Comment (andriisoldatenko) [inline src/lib/deployments/validate/environment/index.unit.test.js:35]: 👍

**Pull request merged**